### PR TITLE
Fix assertEqual's handling of numpy array inputs

### LINKED ISSUE #47948: assertEqual does not compare scalar numpy.ndarray with torch.Tensor correctly
## 🐛 Bug

Add a new test

```python
    def test_assertEqual(self, device):
        a = make_tensor((), device, torch.int32, low=-5, high=5)
        self.assertEqual(a.cpu().numpy(), a, rtol=0, atol=0)
```

run this test, it will fail with something like
```
AssertionError: array(-4, dtype=int32) != tensor(-4, device='cuda:0', dtype=torch.int32)
```

It will not fail if you have a non-scalar tensor `a`

cc @ezyang @gchanan @zou3519 @bdhirsh

### PR DESCRIPTION
**Pull request by kurtamohler**
Fixes #47948

### CODE REVIEW

File: torch/testing/_internal/common_utils.py [hunk 1]
```diff
@@ -1147,6 +1147,14 @@ def assertEqual(self, x, y, msg: Optional[str] = None, *,
                 assert debug_msg is not None
                 msg = "Scalars failed to compare as equal! " + debug_msg
             self.assertTrue(result, msg=msg)
+        # Tensor x Numpy array
```
Comment (mruberry) [inline torch/testing/_internal/common_utils.py:1150]: Should we also add a NumPy x NumPy case to be extra thorough?

Comment (kurtamohler) [reply torch/testing/_internal/common_utils.py:1150]: I added a case for Numpy x Numpy in the tests and it seems to be passing as is. Despite that, I could add it here too if you think it's a good idea

Comment (mruberry) [reply torch/testing/_internal/common_utils.py:1150]: Just so I can sleep easier at night. Let's just convert both to tensors.

Comment (kurtamohler) [reply torch/testing/_internal/common_utils.py:1150]: Done

**Review by mruberry — APPROVED** [review]
Thanks @kurtamohler!

**Pull request closed without merging**